docs/usage/PostMigrationCleanup.md

```diff
@@ -0,0 +1,50 @@
+# Post-migration cleanup
+
+If you've finished migrating your apps with Crane and you want to cleanly remove Crane and associated migration resources from the cluster, follow these steps:
+
+
+## Clean up steps
+
+### For _control clusters_ where mig-controller is installed:
+
+1. Login to the cluster
+1. Set `spec.closed: true` on all MigPlans to run cleanup routines for each plan. This will remove migration related resources spread across the cluster.
+    ```sh
+    for migplan in $(oc get migplan -n openshift-migration -o jsonpath='{.items[*].metadata.name}'); do oc -n openshift-migration patch migplan $migplan --type=json --patch '[{ "op": "add", "path": "/spec/closed", "value": "true" }]'; done
+    ```
+1. Wait for each MigPlan status.conditions to reflect that the plan is closed.
+    ```sh
+    watch oc get migplan -n openshift-migration -o json
+    ```
+1. Follow instructions below for common removal steps.
+
+
+### For _all clusters_ where Crane is installed:
+
+1. Login to the cluster
+1. Remove the `openshift-migration` namespace where Crane is installed
+    ```sh
+    oc delete namespace openshift-migration
+    ```
+1. Remove Crane API objects and CRDs
+    ```sh
+    # Remove mig-operator CRD
+    oc delete customresourcedefinition migrationcontrollers.migration.openshift.io
+
+    # Remove mig-controller CRDs
+    oc delete customresourcedefinition directimagemigrations.migration.openshift.io
+    oc delete customresourcedefinition directimagestreammigrations.migration.openshift.io
+    oc delete customresourcedefinition directvolumemigrationprogresses.migration.openshift.io
+    oc delete customresourcedefinition directvolumemigrations.migration.openshift.io
+    oc delete customresourcedefinition miganalytics.migration.openshift.io
+    oc delete customresourcedefinition migclusters.migration.openshift.io
+    oc delete customresourcedefinition mighooks.migration.openshift.io
+    oc delete customresourcedefinition migmigrations.migration.openshift.io
+    oc delete customresourcedefinition migplans.migration.openshift.io
+    oc delete customresourcedefinition migstorages.migration.openshift.io
+    ```
+1. Remove cluster-scoped RBAC resources installed by Crane
+   ```sh
+   oc delete clusterrolebinding migration-controller
+   oc delete clusterrole migration-controller
```

Comment:
For non-olm clusters, the above name is used for those. But we also have `migration-operator` and `migration-velero`.
For olm-installed clusters, though, doesn't olm use generateName or something similar for these?

Reply:
> For non-olm clusters, the above name is used for those. But we also have migration-operator and migration-velero.

I will add these clusterroles to the list

> For olm-installed clusters, though, doesn't olm use generateName or something similar for these?

Didn't recall this detail. I suspect that if we just delete the OLM CSV(part of openshift-migration ns) it should clean up everything that OLM installed.

Reply:
@sseago updated, please take another look.

Reply:
So the clusterroles and clusterrolebindings above are only created for non-olm. The olm-installed ones have autogenerated names -- I remember this because after changing the velero one to migration-velero to avoid clashing with upstream velero, I realized that we didn't have a change for OLM. Confirmed with Jason that OLM generates its own with non-clashing names, although I'm not sure what the implementation is. It sounds like something is deleting them automatically in this case.